transport/server.cc

```diff
@@ -960,6 +960,8 @@ make_result(int16_t stream, messages::result_message& msg, const tracing::trace_
         cql_protocol_version_type version, bool skip_metadata = false);
 
 template<typename Process>
+requires std::same_as<std::invoke_result_t<Process, service::client_state&, distributed<cql3::query_processor>&, request_reader,
+        uint16_t, cql_protocol_version_type, service_permit, tracing::trace_state_ptr, bool, cql3::computed_function_values>, future<cql_server::process_fn_return_type>>
 future<cql_server::result_with_foreign_response_ptr>
 cql_server::connection::process_on_shard(::shared_ptr<messages::result_message::bounce_to_shard> bounce_msg, uint16_t stream, fragmented_temporary_buffer::istream is,
```

Comment:
Out of scope: we should consider detemplating this.

Reply:
Agreed.  I didn't want to touch that because it is out of scope

Reply:
future work captured in commit log in v4